alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultNotificationAccessor.java

```diff
@@ -176,6 +178,12 @@ public void setNotificationsProcessed(List<AlertNotificationModel> notifications
         notificationContentRepository.saveAll(notificationEntities);
     }
 
+    @Override
+    public void setNotificationsProcessedById(Set<Long> notificationIds) {
+        List<AlertNotificationModel> notificationModels = findByIds(new ArrayList<>(notificationIds));
+        setNotificationsProcessed(notificationModels);
```

Comment:
It seems to me that we should be doing the opposite here. Instead of looking up notifications and calling the old method, we should make the old method extract a set of notificationIds and pass them to a new method that makes one call to the DB to set notifications by id.